src/integrations/prefect-dask/prefect_dask/task_runners.py

```diff
@@ -340,38 +447,26 @@ async def _start(self, exit_stack: AsyncExitStack):
                 f"Creating a new Dask cluster with "
                 f"`{to_qualified_name(self.cluster_class)}`"
             )
-            self._connect_to = self._cluster = await exit_stack.enter_async_context(
-                self.cluster_class(asynchronous=True, **self.cluster_kwargs)
+            self._connect_to = self._cluster = exit_stack.enter_context(
+                self.cluster_class(**self.cluster_kwargs)
             )
+
             if self.adapt_kwargs:
-                adapt_response = self._cluster.adapt(**self.adapt_kwargs)
-                if inspect.isawaitable(adapt_response):
-                    await adapt_response
+                maybe_coro = self._cluster.adapt(**self.adapt_kwargs)
+                if inspect.isawaitable(maybe_coro):
+                    run_sync(maybe_coro)
 
-        self._client = await exit_stack.enter_async_context(
-            distributed.Client(
-                self._connect_to, asynchronous=True, **self.client_kwargs
-            )
+        self._client = exit_stack.enter_context(
+            PrefectDistributedClient(self._connect_to, **self.client_kwargs)
         )
 
         if self._client.dashboard_link:
             self.logger.info(
                 f"The Dask dashboard is available at {self._client.dashboard_link}",
             )
 
-    def __getstate__(self):
-        """
-        Allow the `DaskTaskRunner` to be serialized by dropping
-        the `distributed.Client`, which contains locks.
-        Must be deserialized on a dask worker.
-        """
-        data = self.__dict__.copy()
-        data.update({k: None for k in {"_client", "_cluster", "_connect_to"}})
-        return data
+        return self
 
-    def __setstate__(self, data: dict):
-        """
-        Restore the `distributed.Client` by loading the client on a dask worker.
-        """
-        self.__dict__.update(data)
-        self._client = distributed.get_client()
+    def __exit__(self, *args):
```

Comment:
What let you drop the serialization requirement?

Reply:
Previously, a `PrefectFuture` carried an instance of the task runner that submitted the run around with it. We don't have to do that anymore because we directly wrap a future, so we shouldn't be pickling task runners anymore.

Reply:
Nice.